src/main/java/com/nutomic/syncthingandroid/activities/ShareActivity.java

```diff
@@ -0,0 +1,283 @@
+package com.nutomic.syncthingandroid.activities;
+
+import android.app.ProgressDialog;
+import android.content.ContentResolver;
+import android.content.Intent;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.provider.MediaStore;
+import android.text.TextUtils;
+import android.util.Log;
+import android.webkit.MimeTypeMap;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.nutomic.syncthingandroid.R;
+import com.nutomic.syncthingandroid.model.Folder;
+import com.nutomic.syncthingandroid.service.SyncthingService;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.text.DateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Shares incoming files to syncthing folders.
+ */
+public class ShareActivity extends SyncthingActivity
+        implements SyncthingActivity.OnServiceConnectedListener, SyncthingService.OnApiChangeListener {
+
+    private static final String TAG = "ShareActivity";
+
+    @Override
+    public void onApiChange(SyncthingService.State currentState) {
+        if (currentState != SyncthingService.State.ACTIVE || getApi() == null)
+            return;
+
+        List<Folder> folders = getApi().getFolders();
+
+        ArrayAdapter<Folder> adapter = new ArrayAdapter<>(
+                this, android.R.layout.simple_spinner_item, folders);
```

Comment:
I wanted to get rid of one more `.xml` creation. It's fixed very fast: 
```
public String toString() {
        return label != null ? label : id;
    }
```
Still want separate file? I agree, it's more correct way, but implies new file creation, when anyway `toString()` is not used by anybody + it's still rather good return value.